Switch a "buf" declaration from char to unsigned char.

XXX clarify and send upstream

Index: gdm-2.18.1/daemon/misc.c
===================================================================
--- gdm-2.18.1.orig/daemon/misc.c	2007-05-02 18:38:14.000000000 +0200
+++ gdm-2.18.1/daemon/misc.c	2007-05-02 18:38:14.000000000 +0200
@@ -1360,7 +1360,7 @@
 int
 gdm_fdgetc (int fd)
 {
-	char buf[1];
+	unsigned char buf[1];
 	int bytes;
 
 	VE_IGNORE_EINTR (bytes = read (fd, buf, 1));
